lang/cs/Org.Apache.REEF.Common/Telemetry/CountersData.cs

```diff
@@ -81,14 +81,14 @@ internal void Reset()
         /// Convert the counter data into ISet for sink
         /// </summary>
         /// <returns></returns>
-        internal ISet<KeyValuePair<string, string>> GetCounterData()
+        internal IList<KeyValuePair<string, string>> GetCounterData()
         {
-            var set = new HashSet<KeyValuePair<string, string>>();
+            var list = new List<KeyValuePair<string, string>>();
             foreach (var c in _counterMap)
             {
-                set.Add(c.Value.GetKeyValuePair());
+                list.Add(c.Value.GetKeyValuePair());
             }
-            return set;
+            return list;
```

Comment:
The entire body can be shortened to
```C#
return _counterMap.Select(counter => counter.Value.GetKeyValuePair()).ToList();
```
(or even without the final `.ToList()`, if we can return `IEnumerable`)

Reply:
Neat!